sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -1463,6 +1474,37 @@ abstract class DynamicPartitionPruningSuiteBase
       }
     }
   }
+
+  test("SPARK-34637: test DPP side broadcast query stage is created firstly") {
+    withSQLConf(SQLConf.DYNAMIC_PARTITION_PRUNING_REUSE_BROADCAST_ONLY.key -> "true") {
+      val df = sql(
+        """ WITH view1 as (
+          |   SELECT f.store_id FROM fact_stats f WHERE f.units_sold = 70 group by f.store_id
+          | )
+          |
+          | SELECT * FROM view1 v1 join view1 v2 WHERE v1.store_id = v2.store_id
+        """.stripMargin)
+
+      // A possible resulting query plan:
+      // BroadcastHashJoin
+      // +- HashAggregate
+      //    +- ShuffleQueryStage
+      //       +- Exchange
+      //          +- HashAggregate
+      //             +- Filter
+      //                +- FileScan
+      //                      Dynamicpruning Subquery
```

Comment:
Did try to explain some queries locally? If you did you should see how subqueries are displayed. For example, `select 1, (select 2)`:
```
Project [1 AS 1#7, scalar-subquery#6 [] AS scalarsubquery()#9]
:  +- Project [2 AS 2#8]
:     +- OneRowRelation
+- OneRowRelation
```

Reply:
Yes. In this case, it looks like the following:
![image](https://user-images.githubusercontent.com/11972570/117776519-4349f880-b26e-11eb-9dc6-e1ec7bc05425.png)